core-bundle/src/Command/InstallCommand.php

```diff
@@ -239,8 +247,10 @@ private function addInitializePhp()
      */
     private function symlinkTcpdfConfig()
     {
+        $relPath = $this->fs->makePathRelative($this->bundlesMeta['ContaoCoreBundle']['path'], $this->rootDir);
```

Comment:
```suggestion
        $relPath = rtrim($this->fs->makePathRelative($this->bundlesMeta['ContaoCoreBundle']['path'], $this->rootDir), '/');
```

This is how Symfony uses the `makePathRelative()` method, too: 
https://github.com/symfony/symfony/blob/f194602c3077b6832027f754350f72a1af74927d/src/Symfony/Bundle/FrameworkBundle/CacheWarmer/TemplatePathsCacheWarmer.php#L49

We should not use an external dependency for simple things like this, even if the package is already there, because another package depends on it. Someday the other package may remove its dependency and then Contao will have an additional dependency only to avoid an `rtrim()` call.

Also, since you did not add `"webmozart/path-util"` in the `composer.json` file, we are now relying on the dependencies of our dependency, which is an anti-pattern that must be avoided. Otherwise, if `contao/image` decided to remove its dependency, Contao would break.

Reply:
> Also, since you did not add `"webmozart/path-util"` in the `composer.json` file

It is already there: https://github.com/contao/contao/blob/c2d13af79d35aa1bff8a16472f0546b844e6b6d4/composer.json#L76

I am in favor of using such a dependency. Contao 4 had lots of problems with paths in the past (which mostly I experienced, since I am on Windows 😉 and they probably could have been avoid, if such a dependency was used for any kind of path building.

And since the dependency is already there in the `contao/core-bundle` directly, why not use it?

Reply:
It is no longer a dependency in Contao 4.9 (see https://github.com/contao/contao/blob/4.9/core-bundle/composer.json).

Reply:
I know, but it is in Contao 4.4. And this PR is only for Contao 4.4, it is not needed for Contao 4.9.

Reply:
That is even worse, because it will complicate merging the changes upstream. It will cause merge conflicts that need to be resolved manually and often this is forgotten (like with the `loginCount` -> `loginAttempts` change).

Reply:
Well, with or without this, this cannot be merged upstream, since `tcpdf.php` does not exist anymore in 4.9. I don't know how this is usually handled?

Reply:
It will become a merge conflict that I will have to fix manually.

Reply:
But that's independent of the `webmozart/path-util` dependency. The only thing to take core of while manually merging is to not use the `use` statement upstream - and even that you don't have to remember yourself, since the CS fixer will remove it for you.

Regarding the other changes: in order to be able to more easily merge this upstream, we could look for an alternative approach where we don't have to inject `%kernel.bundles_meta%`.

Reply:
No, using `%kernel.bundles_meta%` is correct.

I just don't understand why we cannot use the `makePathRelative()` method the same way that Symfony does (and we already do e.g. in the `MigrateCommand`, `ImagesLoader` and `SymlinkUtil` classes). Also, there are a lot of places in the core where we join paths and are not using `Path::join()` yet; should we not adjust these as well?

Reply:
Webmozart's `Path` and Symfony's `Filesystem` are two very different components with only a small overlap: `Fileystem` mainly provides wrappers around PHP's file functions to make them behave similar across systems while `Path` is a collection of functions to aid with altering paths (strings). 

I'm using `Path` in #1753 as well because it allows going from relative to absolute paths, getting the file extension, joining paths, normalizing/canonicalizing (think 'a/b/../' paths and 'c:\a\b'). It's a small package that does exactly what it should. I don't get why we should get rid of it? 

Your examples are exactly why I would use it: Instead of manually rtimming + applying slashes all over the place - which is error prone - you could simply use `Path::join`.

We're currently also requiring it implicitly via `contao/image` anyway.

Reply:
> I just don't understand why we cannot use the `makePathRelative()` method the same way that Symfony does (and we already do e.g. in the `MigrateCommand`, `ImagesLoader` and `SymlinkUtil` classes). Also, there are a lot of places in the core where we join paths and are not using `Path::join()` yet; should we not adjust these as well?

I would say yes, we should use `Path` everywhere, in order to prevent errors in the future. But that's outside the scope of this PR imo. In this PR I am simply using something that is already present.